packages/terra-dropdown-button/src/DropdownButton.jsx

```diff
@@ -57,29 +57,62 @@ class DropdownButton extends React.Component {
     this.handleDropdownRequestClose = this.handleDropdownRequestClose.bind(this);
     this.handleKeyDown = this.handleKeyDown.bind(this);
     this.handleKeyUp = this.handleKeyUp.bind(this);
+    this.setButtonNode = this.setButtonNode.bind(this);
+    this.getButtonNode = this.getButtonNode.bind(this);
+    this.setListNode = this.setListNode.bind(this);
+    this.state = { isOpen: false, isActive: false, openedViaKeyboard: false };
+  }
+
+  setListNode(element) {
+    this.dropdownList = element;
+  }
 
-    this.state = { isOpen: false, isActive: false };
+  setButtonNode(node) {
+    this.buttonNode = node;
   }
 
-  handleDropdownButtonClick() {
+  getButtonNode() {
+    return this.buttonNode;
+  }
+
+  handleDropdownButtonClick(event) {
+    if (this.state.isOpen) {
+      this.setState({ openedViaKeyboard: false });
+    }
     this.setState(prevState => ({ isOpen: !prevState.isOpen }));
+    // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
+    // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
+    // This will put focus on the button when clicked.
+    event.currentTarget.focus();
   }
 
   handleDropdownRequestClose(callback) {
-    this.setState({ isOpen: false }, typeof callback === 'function' ? callback : undefined);
+    const onSelectCallback = typeof callback === 'function' ? callback : undefined;
+    this.setState({ isOpen: false, openedViaKeyboard: false, isActive: false }, onSelectCallback);
   }
 
-  /*
-    In FireFox active styles don't get applied on space
-   */
   handleKeyDown(event) {
-    if (event.keyCode === KeyCode.KEY_SPACE) {
-      this.setState({ isActive: true });
+    if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
+      // In FireFox active styles don't get applied on space
+      this.setState({ isActive: true, openedViaKeyboard: true });
+    } else if ((event.keyCode === KeyCode.KEY_DOWN || event.keyCode === KeyCode.KEY_TAB) && this.state.isOpen && !this.state.openedViaKeyboard) {
```

Comment:
We may need to get UX guidance on the use of the tab key to shift focus the dropdown. Using the example mentioned on the issue for reference the behavior is different.

(Mouse click to open dropdown and then press tab)

https://www.w3.org/TR/wai-aria-practices-1.1/examples/menubar/menubar-2/menubar-2.html

Reply:
restored tab key to previous functionality of closing dropdown on Tab Key press. 

Thank you for pointing it out, I was misguided by initial WDIO test code changes  which had Tab key press logic implemented to access list items in both split button and dropdown button. 
Reverted WDIO changes to Previous code and replaced Tab with ArrowDown key press to access list items when opened with mouse click.